crates/cli/src/subcommands/init.rs

```diff
@@ -1,28 +1,979 @@
+        .interact_text()?
+        .trim()
+        .to_string();
+
+    Ok(name)
+}
+
+async fn get_project_path(args: &ArgMatches, project_name: &str, is_interactive: bool) -> anyhow::Result<PathBuf> {
+    if let Some(path) = args.get_one::<PathBuf>("project-path") {
+        if is_interactive {
+            println!("{} {}", "Project path:".bold(), path.display());
+        }
+        return Ok(path.clone());
+    }
+
+    if !is_interactive {
+        return Ok(PathBuf::from(slugify(project_name)));
+    }
+
+    let theme = ColorfulTheme::default();
+    let path_str = Input::with_theme(&theme)
+        .with_prompt("Project path")
+        .default(format!("./{}", slugify(project_name)))
+        .validate_with(|input: &String| -> Result<(), String> {
+            if input.trim().is_empty() {
+                return Err("Project path cannot be empty".to_string());
+            }
+
+            let path = Path::new(input);
+            if path.exists() {
+                if !path.is_dir() {
+                    return Err(format!("A file exists at '{}'. Please choose a different path.", input));
+                }
+                match std::fs::read_dir(path) {
+                    Ok(entries) => {
+                        if entries.count() > 0 {
+                            return Err(format!(
+                                "Directory '{}' already exists and is not empty. Please choose a different path.",
+                                input
+                            ));
+                        }
+                    }
+                    Err(_) => {
+                        return Err(format!(
+                            "Cannot access directory '{}'. Please choose a different path.",
+                            input
+                        ));
+                    }
+                }
+            }
+            Ok(())
+        })
+        .interact_text()?
+        .trim()
+        .to_string();
+
+    Ok(PathBuf::from(path_str))
+}
+
+fn create_template_config_from_template_str(
+    project_name: String,
+    project_path: PathBuf,
+    template_str: &str,
+    templates: &[TemplateDefinition],
+) -> anyhow::Result<TemplateConfig> {
+    if let Some(template) = templates.iter().find(|t| t.id == template_str) {
+        // Builtin template
+        Ok(TemplateConfig {
+            project_name,
+            project_path,
+            template_type: TemplateType::Builtin,
+            server_lang: parse_server_lang(&template.server_lang)?,
+            client_lang: parse_client_lang(&template.client_lang)?,
+            github_repo: None,
+            template_def: Some(template.clone()),
+            use_local: true,
+        })
+    } else {
+        // GitHub template
+        Ok(TemplateConfig {
+            project_name,
+            project_path,
+            template_type: TemplateType::GitHub,
+            server_lang: None,
+            client_lang: None,
+            github_repo: Some(template_str.to_string()),
```

Comment:
Should we check to see if this looks like a URL here? Like if someone just mistypes a template name we will end up here too no?